src/lib.rs

```diff
@@ -1,287 +1,340 @@
-    let spinner = Spinner {
-        still_spinning,
-        msg,
-    };
-    return spinner;
+
+/// Color for spinner.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[non_exhaustive]
+pub enum Color {
+    Blue,
+    Green,
+    Red,
+    Yellow,
+    Cyan,
+    White,
 }
 
 impl Spinner {
+    /// Create a new spinner.
+    ///
+    /// # Arguments
+    ///
+    /// * `spinner_type` - The spinner to use.
+    /// * `msg` - The message to display.
+    /// * `color` - The color of the spinner.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use spinoff::*;
+    /// # use std::thread::sleep;
+    /// # use std::time::Duration;
+    /// #
+    /// let sp = Spinner::new(Spinners::Dots, "Hello", Some(Color::Blue));
+    /// sleep(Duration::from_millis(800));
+    /// sp.clear();
+    /// ```
+    ///
+    /// # Notes
+    ///
+    /// * The spinner immediately starts spinning upon creation.
+    ///
+    pub fn new<T>(spinner_type: Spinners, msg: T, color: Option<Color>) -> Self
+    where
+        T: Into<Cow<'static, str>>,
+    {
+        let still_spinning = Arc::new(AtomicBool::new(true));
+
+        // Gain ownership of the message for the thread to use
+        let msg = msg.into();
+
+        // We use atomic bools to make the thread stop itself when the `spinner.stop()` method is called.
+        let handle = thread::spawn({
+            // Clone the atomic bool so that we can use it in the thread and return the original one later.
+            let still_spinning = Arc::clone(&still_spinning);
+
+            let msg = msg.clone();
+
+            move || {
+                let spinner_data = SPINNER_FRAMES.get(&spinner_type).unwrap();
+
+                let stdout = stdout();
+
+                // Iterate over all the frames of the spinner while the atomic bool is true.
+                let frames = spinner_data
+                    .frames
+                    .iter()
+                    .cycle()
+                    .take_while(|_| still_spinning.load(std::sync::atomic::Ordering::Relaxed));
+
+                let mut last_length = 0;
+
+                for frame in frames {
+                    // This is dependant on the stdout attached is a terminal that respects \r
+                    let mut stdout_lock = stdout.lock();
+                    let frame_str = format!(" {} {}", init_color(color, frame.to_string()), msg);
+                    delete_last_line(last_length);
+                    last_length = frame_str.bytes().len();
+                    write!(stdout_lock, "{}", frame_str).unwrap();
+                    stdout_lock.flush().unwrap();
+                    drop(stdout_lock);
+
+                    thread::sleep(std::time::Duration::from_millis(
+                        spinner_data.interval as u64,
+                    ));
+                }
+
+                delete_last_line(last_length);
```

Comment:
```suggestion
              
```
I think it would be better to delete this line and move it into `stop_spinner_thread` instead. It would also fix my previous comment.